shoov/modules/custom/shoov_ci_build/shoov_ci_build.module

```diff
@@ -58,6 +69,81 @@ function  shoov_ci_build_trigger_ci_build_item($message, $ajax = 'nojs') {
 }
 
 /**
+ * Check permissions to un/subscribe account to the CI Build.
+ *
+ * @param $ci_build
+ *  The CI Build node object
+ * @param $account
+ *  The user object.
+ * @param $token
+ *  The access token
+ *
+ * @return bool
+ *  Returns TRUE if user has permissions to un/subscribe account to the CI Build.
+ */
+function shoov_ci_build_subscribe_user_check_access($ci_build, $account, $token) {
+  if (!user_access('administer flags')) {
+    // User doesn't have permissions to un/subscribe account.
+    return FALSE;
+  }
+
+  if (md5($ci_build->nid . '+' . $account->uid) != $token) {
```

Comment:
protection, as the menu callback is not related to the view - it can be called directly.